Rubberduck.CodeAnalysis/Inspections/Concrete/AssignedByValParameterInspection.cs

```diff
@@ -48,12 +48,41 @@ protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
                 .Cast<ParameterDeclaration>()
                 .Where(item => !item.IsByRef 
                     && !item.IsIgnoringInspectionResultFor(AnnotationName)
-                    && item.References.Any(reference => reference.IsAssignment));
+                    && item.References.Any(IsAssignmentToDeclaration));
 
             return parameters
                 .Select(param => new DeclarationInspectionResult(this,
                                                       string.Format(InspectionResults.AssignedByValParameterInspection, param.IdentifierName),
                                                       param));
         }
+
+        private static bool IsAssignmentToDeclaration(IdentifierReference reference)
+        {
+            if (!reference.IsAssignment)
+            {
+                return false;
+            }
+
+            if (reference.IsSetAssignment)
+            {
+                return true;
+            }
+
+            var declaration = reference.Declaration;
+            if (declaration == null)
+            {
+                return false;
+            }
+
+            if (declaration.IsObject)
+            {
+                //This can only be legal with a default member access.
+                return false;
```

Comment:
This will probably have to stay like this, even if we resolve implicit default member calls, since we cannot get all. (If the type is `Object`, we cannot know the default member.)

Still, I will add a todo in case we can get rid of the assignment flags.

Reply:
Done